src/classes/GE_PaymentServices.cls

```diff
@@ -0,0 +1,144 @@
+/*
+    Copyright (c) 2020, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group HGE
+* @group-content
+* @description Class holds payment services related methods.
+*/
+public with sharing class GE_PaymentServices {
+
+    String paymentMethodToken = null;
+
+    public GE_PaymentServices(String token) {
+        this.paymentMethodToken = token;
+    }
+
+    public HttpResponse makePurchaseCall() {
+        return makePurchaseCall(this.paymentMethodToken);
+    }
+
+    public HttpResponse makePurchaseCall(String token) {
+        // Create http request...
+        Http http = new Http();
+        HttpRequest request = new HttpRequest();
+        request.setEndpoint(getEndpointRoot() + getEndpoint(ElevateEndpoints.PURCHASE));
+        request.setMethod('POST');
+        request.setHeader('Content-Type', 'application/json;charset=UTF-8');
+        request.setHeader('jwttoken', getJwt());
+        request.setHeader('sfdo.merchantids', getMerchantIds());
+        request.setHeader('sfdo.username', getUsername());
+
+        // Set the body as a JSON object...
```

Comment:
@psadie-sfdo I think it might be beneficial if we create a class(maybe an inner class) for the body of the request and maybe call the `JSON.serialize(object)` on the object to help with building the JSON body string instead of building manually.

Reply:
That's a great idea! I wonder if we could make that update in the work item I have now that's focused on making this purchase call. If only to get this merged in and kind of level set in our feature branch. Happy to make that update here if we feel strongly about it now though.

Reply:
That sounds good. I don't think this has to be done in this branch if you can implement in the your current WI. But we can open up this decision to the team. Thoughts? @vmartinov-sforg   @ptewson-sfdo @kenlewis-sfdo